sql/data_sql_warehouse.go

```diff
@@ -31,11 +31,35 @@ func DataSourceWarehouse() *schema.Resource {
 
 	return common.DataResource(SQLWarehouseInfo{}, func(ctx context.Context, e interface{}, c *common.DatabricksClient) error {
 		data := e.(*SQLWarehouseInfo)
-		err := c.Get(ctx, fmt.Sprintf("/sql/warehouses/%s", data.ID), nil, data)
+		var id string
```

Comment:
Hm, this is interesting. Maybe I'm misunderstanding the API, but I think that for either warehouse ID or name, you just need to make a single API call to https://docs.databricks.com/api/workspace/datasources/list and can filter the result list based on the "name" or "warehouse_id" fields.

Reply:
We need to have `Get` call anyway because the data sources APIs don't have all necessary information about warehouse that data source returns

Reply:
I see. Then maybe one small suggestion would be to do these requests in the opposite order. That way you could get the datasource ID and warehouse ID from the data sources API, filtering by warehouse name/ID, then look up the warehouse by ID.